tests/integration/test_logcollector/test_configuration/test_basic_configuration_out_format.py

```diff
@@ -0,0 +1,165 @@
+     'target_out_format': 'custom_socket', 'valid_value': True},
+    {'socket_name': 'custom_socket', 'socket_path': '/var/log/messages', 'location': "/tmp/testing.log",
+     'log_format': 'syslog', 'target': 'custom_socket', 'out_format': '$(timestamp)',
+     'target_out_format': 'custom_socket', 'valid_value': True},
+    {'socket_name': 'custom_socket', 'socket_path': '/var/log/messages', 'location': "/tmp/testing.log",
+     'log_format': 'syslog', 'target': 'custom_socket', 'out_format': '$(hostname)',
+     'target_out_format': 'custom_socket', 'valid_value': True},
+    {'socket_name': 'custom_socket', 'socket_path': '/var/log/messages', 'location': "/tmp/testing.log",
+     'log_format': 'syslog', 'target': 'custom_socket', 'out_format': '$(host_ip)',
+     'target_out_format': 'custom_socket', 'valid_value': True},
+    {'socket_name': 'custom_socket', 'socket_path': '/var/log/messages', 'location': "/tmp/testing.log",
+     'log_format': 'syslog', 'target': 'custom_socket', 'out_format': '$(host_ip)',
+     'target_out_format': 'no_defined_custom_socket', 'valid_value': False},
+]
+
+configurations = load_wazuh_configurations(configurations_path, __name__,
+                                           params=parameters,
+                                           metadata=metadata)
+configuration_ids = \
+    [
+        f"{x['LOG_FORMAT'], x['TARGET'], x['SOCKET_NAME'], x['LOCATION'], x['SOCKET_PATH'], x['TARGET_OUT_FORMAT'], x['OUT_FORMAT']}"
+        for x in parameters]
+
+
+def check_configuration_out_format_valid(cfg):
+    """Check if the Wazuh run correctly with the specified out format field.
+
+    Ensure logcollector allows the specified out format attribute.
+
+    Args:
+        cfg (dict): Dictionary with the localfile configuration.
+
+    Raises:
+        TimeoutError: If the callback for socket target is not generated.
+        AssertError: In case of a server instance, the API response is different that the real configuration.
+    """
+    log_callback = logcollector.callback_socket_target(cfg['location'], cfg['target'], prefix=prefix)
+    wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_TARGET_SOCKET)
+
+    if wazuh_component == 'wazuh-manager':
+        real_configuration = dict((key, cfg[key]) for key in ('location', 'target', 'log_format'))
+        real_configuration['out_format'] = {'target': cfg['target_out_format'], 'item': cfg['out_format']}
+        api.wait_until_api_ready()
+        api.compare_config_api_response([real_configuration], 'localfile')
+
+
+def check_configuration_out_format_invalid(cfg):
+    """Check if the Wazuh fails because a invalid out format configuration value.
+
+    Args:
+        cfg (dict): Dictionary with the localfile configuration.
+
+    Raises:
+        TimeoutError: If error callback are not generated.
+    """
```

Comment:
```suggestion
    """Check if the Wazuh fails because an invalid out format configuration value.

    Args:
        cfg (dict): Dictionary with the localfile configuration.

    Raises:
        TimeoutError: If the error callbacks are not generated.
    """
```